channels/applicationinsights-channel-js/src/SendBuffer.ts

```diff
@@ -2,15 +2,20 @@ import dynamicProto from "@microsoft/dynamicproto-js";
 import { utlGetSessionStorage, utlSetSessionStorage } from "@microsoft/applicationinsights-common";
 import {
     IDiagnosticLogger, _eInternalMessageId, _throwInternal, arrForEach, arrIndexOf, dumpObj, eLoggingSeverity, getExceptionName, getJSON,
-    isArray, isFunction, isString
+    isArray, isFunction, isNullOrUndefined, isString
 } from "@microsoft/applicationinsights-core-js";
-import { ISenderConfig } from "./Interfaces";
+import { IInternalStorageItem, ISenderConfig } from "./Interfaces";
 
+/**
+ * Before 3.1.3, payload only allow string
+ * After 3.1.3,  IInternalStorageItem is accepted
+ */
 export interface ISendBuffer {
+
     /**
      * Enqueue the payload
      */
-    enqueue: (payload: string) => void;
+    enqueue: (payload: IInternalStorageItem) => void;
```

Comment:
sendBuffer accepts IInternalStorageItem instead of string . (previous versions will continue using string, since buffer key is different, so there "should" have no conflicts)